pkg/controller/imagebuild/component/builddispatcher.go

```diff
@@ -104,24 +107,39 @@ func (c *BuildDispatcherComponent) Reconcile(ctx *core.Context) (ctrl.Result, er
 	}(configDir)
 
 	validateCredsSeg := txn.StartSegment("credentials-validate")
-	if err = credentials.Verify(ctx, configDir); err != nil {
+
+	insecureRegistries := make([]string, 0)
+	for reg, opts := range c.cfg.Registries {
+		if opts.Insecure || opts.HTTP {
+			insecureRegistries = append(insecureRegistries, reg)
+		}
+	}
+
+	buildLog.Info("Validating registry credentials")
+	if err = credentials.Verify(ctx, configDir, insecureRegistries); err != nil {
 		txn.NoticeError(newrelic.Error{
 			Message: err.Error(),
 			Class:   "CredentialsValidateError",
 		})
+
+		buildLog.Error(err, fmt.Sprintf("Failed to validate registry credentials: %s", err.Error()))
```

Comment:
This prints an error like this when the registry is plainhttp without a valid config:
```
Jul 14 2022 18:37:13 -0700 | Validating registry credentials
Jul 14 2022 18:37:13 -0700 | Failed to validate registry credentials: "10.0.0.8:5001" client credentials are invalid: Get "https://10.0.0.8:5001/v2/": http: server gave HTTP response to HTTPS client  
```